### PR TITLE
WebGPU support

### PR DESCRIPTION
**Pull request by slimbuck**
This PR makes the following changes:
- 🎉🎉🎉 add support for webgpu rendering 🎉🎉🎉
- update WebXR AR object placement mode to support two-finger rotation, vibrate on placement
- add ability to frame selected node using 'F' key
- update to engine v1.66
- many small tweaks

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import {
-    platform,
-    WebglGraphicsDevice,
+    // platform,
+    // WebglGraphicsDevice,
```
Comment (willeastcott) [inline src/app.ts:3]: Can we just delete these?

File: src/index.tsx [hunk 2]
```diff
@@ -130,20 +145,17 @@ const observerData: ObserverData = {
...
     },
+    runtime: {
+        deviceType: ''
+    },
     morphs: null,
     spinner: false,
     error: null,
     xrSupported: false,
-    xrActive: false
+    xrActive: false,
```
Comment (willeastcott) [inline src/index.tsx:155]: I thought we disallowed trailing commas in object definitions. Is ESLint not running on tsx files in this project?

File: src/xr-mode.ts [hunk 3]
```diff
@@ -1,295 +1,462 @@
...
-    }
+        createModelHandler(this);
 
-    // start the XR session
-    start() {
-        if (this.app.xr.isAvailable(XRTYPE_AR)) {
-            this.handlers.starting();
-            this.app.xr.start(this.camera.camera, XRTYPE_AR, XRSPACE_LOCAL, {
-                callback: (err: Error | null) => {
+        // perform an asynchronous ray interesection test given a view-space ray
```
Comment (willeastcott) [inline src/xr-mode.ts:375]:
```suggestion
        // perform an asynchronous ray intersection test given a view-space ray
```

**Review by willeastcott — APPROVED** [review]
Really fantastic! Viewer v4 is gonna ruuuuule! 🚀

File: src/app.ts (hunk 1 above)
Comment (slimbuck) [reply src/app.ts:3]: yes, thanks

File: src/index.tsx (hunk 2 above)
Comment (slimbuck) [reply src/index.tsx:155]: I guess not? @playcanvas/eslint-config/index.js:

Screenshot 2023-10-09 at 11 32 34

**Pull request merged**